include/internal/HMDNSResult.h

```diff
@@ -67,30 +70,34 @@ class HMDNSResult
 	/*!
 	     Start the DNS resolution for this result.
 	     Sets the internal state to HM_CHECK_IN_PROGRESS.
+             \param version of the query.
 	     \return the HMTimeStamp with the check timeout value. (To determine when to reschedule in case of timeout.)
 	 */
-	HMTimeStamp startQuery();
+	HMTimeStamp startQuery(uint32_t version=0);
 
 	//! Finish the DNS resolution for this result.
 	/*!
 	     Finish the DNS resolution for this result. Set the internal state to either HM_CHECK_IDLE or HM_CHECK_FAILED.
 	     \param true if the lookup was a success.
+             \param version of the query.
 	 */
 	void finishQuery(bool success);
 
 	//! Determine if the DNS query needs to be run now.
 	/*!
 	     Determine if the DNS query needs to be run now.
+             \param version of the current state.
 	     \return true if the check is expired and needs to be run now.
 	 */
-	bool queryNeeded() const;
+	bool queryNeeded(uint32_t version=0) const;
```

Comment:
Can you create a constant HM_DEFAULT_STATE_VERSION = 0 in HMConstants.h and use it